WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/lists/sections/BaseStatsUseCaseTest.kt

```diff
@@ -79,15 +79,15 @@ class BaseStatsUseCaseTest : BaseUnitTest() {
         advanceUntilIdle()
 
         assertThat(result.size).isEqualTo(1)
-        assertData(0, localData)
+        assertData(0, remoteData)
```

Comment:
We should also update test test name.

Reply:
Name is actually still accurate. I ran the test to confirm. The problem is actually threading (advanceUntilIdle). When you call `block.fetch`, 2-4 jobs are potentially created,  one for each call to `updateState`. But the threads/jobs created don't immediately run until we call `advanceUntilIdle`.  `BaseStatsUseCase ` would need to be re-written to more accurately reflect what would happen when we have more exact control of each job/thread. At the moment I'd need to make even more changes to improve the test. More than I think should be in scope here. Does this make sense?

Reply:
Fair enough.